packages/avatar/contracts/nft-collection/NFTCollection.sol

```diff
@@ -745,6 +773,7 @@ INFTCollection
 
     /**
      * @notice return ERC20 contract through which the minting will be done (approveAndCall)
+     * @return the address of the token that is allowed to do a call to mint
```

Comment:
Maybe simplify with:

`the address of the token that is allowed to execute mint function`?